src/native-stack/views/NativeStackView.tsx

```diff
@@ -423,6 +425,25 @@ export default function NativeStackView(props: Props) {
   );
 }
 
+function getStatusBarHeight(dimensions: Rect, isStatusBarTranslucent: boolean) {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const topInset = useSafeAreaInsets().top;
```

Comment:
We can pass it as a argument to the function to avoid breaking eslint / react rules.

Reply:
Resolved in https://github.com/software-mansion/react-native-screens/pull/1873/commits/a19dbd98d529f7af5602f28d628348ce01f1db94.